### PR TITLE
[stdlib] Replacing M_PI with .pi

### PR DESCRIPTION
**Pull request by moiseev**
`M_PI` has recently been deprecated. Replacing it's usages to avoid compilation warnings.

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@swift-ci Please smoke test and merge

File: stdlib/public/SDK/Foundation/AffineTransform.swift [hunk 1]
```diff
@@ -124,7 +124,7 @@ public struct AffineTransform : ReferenceConvertible, Hashable, CustomStringConv
          [    0       0    1 ]
      */
     public init(rotationByDegrees angle: CGFloat) {
-        let α = Double(angle) * M_PI / 180.0
+        let α = Double(angle) * .pi / 180.0
         self.init(rotationByRadians: CGFloat(α))
```
Comment (stephentyrone) [inline stdlib/public/SDK/Foundation/AffineTransform.swift]: These could be cleaned up further, I think; there's no long any need for the conversions.

    self.init(rotationByRadians: angle * .pi / 180)

Possibly outside the scope of this change.

File: test/stdlib/TestAffineTransform.swift [hunk 2]
```diff
@@ -169,19 +169,19 @@ class TestAffineTransform : TestAffineTransformSuper {
         checkPointTransformation(noop, point: point, expectedPoint: point)
         
         var tenEighty = AffineTransform.identity
-        tenEighty.rotate(byRadians: CGFloat(6 * M_PI))
+        tenEighty.rotate(byRadians: CGFloat(6 * Double.pi))
```
Comment (stephentyrone) [inline test/stdlib/TestAffineTransform.swift]: Similarly, these could just be `tenEighty.rotate(byRadians: 6 * .pi)`.

**Review by stephentyrone — COMMENTED** [review]
A few simplifications are possible because CGFloat also has `.pi`, whereas M_PI was Double. Those may be outside the scope of this change, however.

File: stdlib/public/SDK/Foundation/AffineTransform.swift (hunk 1 above)
Comment (moiseev) [inline stdlib/public/SDK/Foundation/AffineTransform.swift]: Oh. right! `CGFloat` conforms to the same protocol. Makes perfect sense. Thanks!

**Comment by moiseev** [comment]
@swift-ci Please test and merge

**Pull request merged**